Benchmarks don't compile

Trying to compile benchmarks of current master with either stable or nightly fails with: 

```
   Compiling hecs v0.2.7 (/home/svenstaro/src/hecs)
error[E0277]: `hecs::query::QueryBorrow<'_, (&mut Position, &Velocity)>` is not an iterator
  --> benches/bench.rs:50:32
   |
50 |         for (_, (pos, vel)) in world.query::<(&mut Position, &Velocity)>() {
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `hecs::query::QueryBorrow<'_, (&mut Position, &Velocity)>` is not an iterator
   |
   = help: the trait `std::iter::Iterator` is not implemented for `hecs::query::QueryBorrow<'_, (&mut Position, &Velocity)>`
   = note: required by `std::iter::IntoIterator::into_iter`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `hecs`.
```

What am I doing wrong?

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -32,7 +32,7 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: build
-          args: ${{ matrix.features }}
+          args: --all-targets ${{ matrix.features }}
 
       - uses: actions-rs/cargo@v1
         with:

--- a/benches/bench.rs
+++ b/benches/bench.rs
@@ -47,7 +47,7 @@ fn iterate_100k(b: &mut Bencher) {
         world.spawn((Position(-(i as f32)), Velocity(i as f32)));
     }
     b.iter(|| {
-        for (_, (pos, vel)) in world.query::<(&mut Position, &Velocity)>() {
+        for (_, (pos, vel)) in &mut world.query::<(&mut Position, &Velocity)>() {
             pos.0 += vel.0;
         }
     })

--- a/src/entities.rs
+++ b/src/entities.rs
@@ -1,0 +1,137 @@
+use alloc::vec::Vec;
+use core::fmt;
+#[cfg(feature = "std")]
+use std::error::Error;
+
+/// Lightweight unique ID of an entity
+///
+/// Obtained from `World::spawn`. Can be stored to refer to an entity in the future.
+#[derive(Clone, Copy, Hash, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Entity {
+    pub(crate) generation: u32,
+    pub(crate) id: u32,
+}
+
+impl Entity {
+    /// Convert to a form convenient for passing outside of rust
+    ///
+    /// Only useful for identifying entities within the same instance of an application. Do not use
+    /// for serialization between runs.
+    ///
+    /// No particular structure is guaranteed for the returned bits.
+    pub fn to_bits(self) -> u64 {
+        u64::from(self.generation) << 32 | u64::from(self.id)
+    }
+
+    /// Reconstruct an `Entity` previously destructured with `to_bits`
+    ///
+    /// Only useful when applied to results from `to_bits` in the same instance of an application.
+    pub fn from_bits(bits: u64) -> Self {
+        Self {
+            generation: (bits >> 32) as u32,
+            id: bits as u32,
+        }
+    }
+
+    /// Extract a transiently unique identifier
+    ///
+    /// No two simultaneously-live entities share the same ID, but dead entities' IDs may collide
+    /// with both live and dead entities. Useful for compactly representing entities within a
+    /// specific snapshot of the world, such as when serializing.
+    pub fn id(self) -> u32 {
+        self.id
+    }
+}
+
+impl fmt::Debug for Entity {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}v{}", self.id, self.generation)
+    }
+}
+
+#[derive(Default)]
+pub(crate) struct Entities {
+    pub meta: Vec<EntityMeta>,
+    free: Vec<u32>,
+}
+
+impl Entities {
+    pub fn alloc(&mut self) -> Entity {
+        match self.free.pop() {
+            Some(i) => Entity {
+                generation: self.meta[i as usize].generation,
+                id: i,
+            },
+            None => {
+                let i = self.meta.len() as u32;
+                self.meta.push(EntityMeta {
+                    generation: 0,
+                    location: Location {
+                        archetype: 0,
+                        index: 0,
+                    },
+                });
+                Entity {
+                    generation: 0,
+                    id: i,
+                }
+            }
+        }
+    }
+
+    pub fn free(&mut self, entity: Entity) -> Result<Location, NoSuchEntity> {
+        let meta = &mut self.meta[entity.id as usize];
+        if meta.generation != entity.generation {
+            return Err(NoSuchEntity);
+        }
+        meta.generation += 1;
+        self.free.push(entity.id);
+        Ok(meta.location)
+    }
+
+    pub fn clear(&mut self) {
+        self.meta.clear();
+        self.free.clear();
+    }
+
+    pub fn get_mut(&mut self, entity: Entity) -> Result<&mut Location, NoSuchEntity> {
+        let meta = &mut self.meta[entity.id as usize];
+        if meta.generation != entity.generation {
+            return Err(NoSuchEntity);
+        }
+        Ok(&mut meta.location)
+    }
+
+    pub fn get(&self, entity: Entity) -> Result<Location, NoSuchEntity> {
+        let meta = &self.meta[entity.id as usize];
+        if meta.generation != entity.generation {
+            return Err(NoSuchEntity);
+        }
+        Ok(meta.location)
+    }
+}
+
+#[derive(Copy, Clone)]
+pub(crate) struct EntityMeta {
+    pub generation: u32,
+    pub location: Location,
+}
+
+#[derive(Copy, Clone)]
+pub(crate) struct Location {
+    pub archetype: u32,
+    pub index: u32,
+}
+
+/// Error indicating that no entity with a particular ID exists
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct NoSuchEntity;
+
+impl fmt::Display for NoSuchEntity {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.pad("no such entity")
+    }
+}
+
+#[cfg(feature = "std")]
+impl Error for NoSuchEntity {}

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -62,15 +62,17 @@ macro_rules! smaller_tuples_too {
 mod archetype;
 mod borrow;
 mod bundle;
+mod entities;
 mod entity_builder;
 mod query;
 mod world;
 
 pub use borrow::{EntityRef, Ref, RefMut};
 pub use bundle::{Bundle, DynamicBundle, MissingComponent};
+pub use entities::{Entity, NoSuchEntity};
 pub use entity_builder::{BuiltEntity, EntityBuilder};
 pub use query::{Query, QueryBorrow, QueryIter, With, Without};
-pub use world::{Component, ComponentError, Entity, Iter, NoSuchEntity, World};
+pub use world::{Component, ComponentError, Iter, World};
 
 // Unstable implementation details needed by the macros
 #[doc(hidden)]

--- a/src/query.rs
+++ b/src/query.rs
@@ -16,7 +16,7 @@ use core::marker::PhantomData;
 use core::ptr::NonNull;
 
 use crate::archetype::Archetype;
-use crate::world::EntityMeta;
+use crate::entities::EntityMeta;
 use crate::{Component, Entity};
 
 /// A collection of component types to fetch from a `World`

--- a/src/world.rs
+++ b/src/world.rs
@@ -22,7 +22,11 @@ use std::error::Error;
 use hashbrown::{HashMap, HashSet};
 
 use crate::archetype::Archetype;
-use crate::{Bundle, DynamicBundle, EntityRef, MissingComponent, Query, QueryBorrow, Ref, RefMut};
+use crate::entities::{Entities, EntityMeta};
+use crate::{
+    Bundle, DynamicBundle, Entity, EntityRef, MissingComponent, NoSuchEntity, Query, QueryBorrow,
+    Ref, RefMut,
+};
 
 /// An unordered collection of entities, each having any number of distinctly typed components
 ///
@@ -439,68 +443,9 @@ impl From<MissingComponent> for ComponentError {
     }
 }
 
-/// Error indicating that no entity with a particular ID exists
-#[derive(Debug, Clone, Eq, PartialEq)]
-pub struct NoSuchEntity;
-
-impl fmt::Display for NoSuchEntity {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.pad("no such entity")
-    }
-}
-
-#[cfg(feature = "std")]
-impl Error for NoSuchEntity {}
-
 /// Types that can be components (implemented automatically)
 pub trait Component: Send + Sync + 'static {}
 impl<T: Send + Sync + 'static> Component for T {}
-
-/// Lightweight unique ID of an entity
-///
-/// Obtained from `World::spawn`. Can be stored to refer to an entity in the future.
-#[derive(Clone, Copy, Hash, Eq, Ord, PartialEq, PartialOrd)]
-pub struct Entity {
-    pub(crate) generation: u32,
-    pub(crate) id: u32,
-}
-
-impl Entity {
-    /// Convert to a form convenient for passing outside of rust
-    ///
-    /// Only useful for identifying entities within the same instance of an application. Do not use
-    /// for serialization between runs.
-    ///
-    /// No particular structure is guaranteed for the returned bits.
-    pub fn to_bits(self) -> u64 {
-        u64::from(self.generation) << 32 | u64::from(self.id)
-    }
-
-    /// Reconstruct an `Entity` previously destructured with `to_bits`
-    ///
-    /// Only useful when applied to results from `to_bits` in the same instance of an application.
-    pub fn from_bits(bits: u64) -> Self {
-        Self {
-            generation: (bits >> 32) as u32,
-            id: bits as u32,
-        }
-    }
-
-    /// Extract a transiently unique identifier
-    ///
-    /// No two simultaneously-live entities share the same ID, but dead entities' IDs may collide
-    /// with both live and dead entities. Useful for compactly representing entities within a
-    /// specific snapshot of the world, such as when serializing.
-    pub fn id(self) -> u32 {
-        self.id
-    }
-}
-
-impl fmt::Debug for Entity {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}v{}", self.id, self.generation)
-    }
-}
 
 /// Iterator over all of a world's entities
 pub struct Iter<'a> {
@@ -575,80 +520,6 @@ impl<A: DynamicBundle> core::iter::FromIterator<A> for World {
         world.extend(iter);
         world
     }
-}
-
-#[derive(Default)]
-struct Entities {
-    meta: Vec<EntityMeta>,
-    free: Vec<u32>,
-}
-
-impl Entities {
-    fn alloc(&mut self) -> Entity {
-        match self.free.pop() {
-            Some(i) => Entity {
-                generation: self.meta[i as usize].generation,
-                id: i,
-            },
-            None => {
-                let i = self.meta.len() as u32;
-                self.meta.push(EntityMeta {
-                    generation: 0,
-                    location: Location {
-                        archetype: 0,
-                        index: 0,
-                    },
-                });
-                Entity {
-                    generation: 0,
-                    id: i,
-                }
-            }
-        }
-    }
-
-    fn free(&mut self, entity: Entity) -> Result<Location, NoSuchEntity> {
-        let meta = &mut self.meta[entity.id as usize];
-        if meta.generation != entity.generation {
-            return Err(NoSuchEntity);
-        }
-        meta.generation += 1;
-        self.free.push(entity.id);
-        Ok(meta.location)
-    }
-
-    fn clear(&mut self) {
-        self.meta.clear();
-        self.free.clear();
-    }
-
-    fn get_mut(&mut self, entity: Entity) -> Result<&mut Location, NoSuchEntity> {
-        let meta = &mut self.meta[entity.id as usize];
-        if meta.generation != entity.generation {
-            return Err(NoSuchEntity);
-        }
-        Ok(&mut meta.location)
-    }
-
-    fn get(&self, entity: Entity) -> Result<Location, NoSuchEntity> {
-        let meta = &self.meta[entity.id as usize];
-        if meta.generation != entity.generation {
-            return Err(NoSuchEntity);
-        }
-        Ok(meta.location)
-    }
-}
-
-#[derive(Copy, Clone)]
-pub(crate) struct EntityMeta {
-    pub(crate) generation: u32,
-    location: Location,
-}
-
-#[derive(Copy, Clone)]
-struct Location {
-    archetype: u32,
-    index: u32,
 }
 
 #[cfg(test)]

--- a/tests/tests.rs
+++ b/tests/tests.rs
@@ -32,7 +32,9 @@ fn despawn() {
     let mut world = World::new();
     let e = world.spawn(("abc", 123));
     let f = world.spawn(("def", 456));
+    assert_eq!(world.query::<()>().iter().count(), 2);
     world.despawn(e).unwrap();
+    assert_eq!(world.query::<()>().iter().count(), 1);
     assert!(world.get::<&str>(e).is_err());
     assert!(world.get::<i32>(e).is_err());
     assert_eq!(*world.get::<&str>(f).unwrap(), "def");

